enterprise/admin/views.py

```diff
@@ -62,20 +64,52 @@ def _build_context(self, request, customer_uuid):
         """
         # TODO: pylint acts stupid - find a way around it without suppressing
         enterprise_customer = EnterpriseCustomer.objects.get(uuid=customer_uuid)  # pylint: disable=no-member
-        linked_learners = EnterpriseCustomerUser.objects.filter(enterprise_customer__uuid=enterprise_customer.uuid)
-        pending_linked_learners = PendingEnterpriseCustomerUser.objects.filter(
-            enterprise_customer__uuid=enterprise_customer.uuid
-        )
+        filterstring = self.get_filterstring(request)
+        linked_learners = self.get_enterprise_customer_user_queryset(filterstring, customer_uuid)
+        pending_linked_learners = self.get_pending_queryset(filterstring, customer_uuid)
 
         context = {
             self.ContextParameters.ENTERPRISE_CUSTOMER: enterprise_customer,
             self.ContextParameters.PENDING_LEARNERS: pending_linked_learners,
             self.ContextParameters.LEARNERS: linked_learners,
+            self.ContextParameters.FILTERSTRING_NAME: filterstring,
         }
         context.update(admin.site.each_context(request))
         context.update(self._build_admin_context(request, enterprise_customer))
         return context
 
+    def get_filterstring(self, request):
+        """
+        Retrieve the search querystring from the GET parameters.
+        """
+        return request.GET.get('q', None)
+
+    def get_enterprise_customer_user_queryset(self, filterstring, customer_uuid):
+        """
+        Get the list of EnterpriseCustomerUsers we want to render.
+        """
+        learners = EnterpriseCustomerUser.objects.filter(enterprise_customer__uuid=customer_uuid)
+        if filterstring is None:
+            return learners
+        user_ids = learners.values_list('user_id', flat=True)
+        allowed_users = User.objects.filter(
+            Q(pk__in=user_ids),
+            Q(email__icontains=filterstring) | Q(username__icontains=filterstring)
+        )
+        allowed_user_ids = allowed_users.values_list('pk', flat=True)
+        return learners.filter(user_id__in=allowed_user_ids)
+
+    def get_pending_queryset(self, filterstring, customer_uuid):
+        """
+        Get the list of PendingEnterpriseCustomerUsers we want to render.
+        """
+        base_queryset = PendingEnterpriseCustomerUser.objects.filter(
+            enterprise_customer__uuid=customer_uuid
+        )
+        if filterstring is None:
+            return base_queryset
+        return base_queryset.filter(user_email__icontains=filterstring)
```

Comment:
This is the nittiest nit, which you are welcome to ignore: I personally find the following form more clear:
```python
queryset = PendingEnterpriseCustomerUser.objects.filter(
    enterprise_customer__uuid=customer_uuid
)
if filterstring:
    queryset = queryset.filter(user_email__icontains=filterstring)
return queryset
```
(Also, in the future, if a second way of filtering is added, with the above form it's easy to add in additional filters, but with the short-circuit approach you have now, a refactor is required).